File: Assets/RCIM/Plugins/Android/RCUnityIM.androidlib/src/cn/rongcloud/im/wrapper/unity/RCIMConnectCallbackProxy.java
package cn.rongcloud.im.wrapper.unity;

import cn.rongcloud.im.wrapper.callback.IRCIMIWConnectCallback;
import cn.rongcloud.im.wrapper.callback.RCIMIWConnectCallback;

public class RCIMConnectCallbackProxy extends RCIMIWConnectCallback {
  IRCIMIWConnectCallback cb;

  public RCIMConnectCallbackProxy(IRCIMIWConnectCallback listener) {
    this.cb = listener;
  }

  public void onDatabaseOpened(int code) {
    cb.onDatabaseOpened(code);
  }

  public void onConnected(int code, String userId) {
    cb.onConnected(code, userId);
  }
}

File: Assets/RCIM/Scripts/Android/RCIMConnectCallbackProxy.cs
#if UNITY_ANDROID
using System;
using System.Collections.Generic;
using System.Runtime.InteropServices;
using UnityEngine;
using AOT;

namespace cn_rongcloud_im_unity
{
    public class RCIMConnectCallbackProxy : AndroidJavaProxy
    {
        RCIMConnectListener listener;
    
        public RCIMConnectCallbackProxy(RCIMConnectListener listener)
            : base("cn.rongcloud.im.wrapper.callback.IRCIMIWConnectCallback")
        {
            this.listener = listener;
        }
    
        public void onConnected(int code, string userId)
        {
            if (listener != null)
                listener.OnConnected(code, userId);
        }
    
        public void onDatabaseOpened(int code)
        {
            if (listener != null)
                listener.OnDatabaseOpened(code);
        }
    }
}
#endif
